golang/go-cloud-run-hello-world/cloudbuild.yaml

```diff
@@ -0,0 +1,67 @@
+steps:
+
+- id: 'Build Container Image'
+  name: 'gcr.io/cloud-builders/docker:latest'
+  args: ['build', '-t', 'gcr.io/${PROJECT_ID}/${_SERVICE_TAG}:${COMMIT_SHA}', '.']
+
+- id: 'Push Container Image'
+  name: 'gcr.io/cloud-builders/docker:latest'
+  args: ['push', 'gcr.io/${PROJECT_ID}/${_SERVICE_TAG}:${COMMIT_SHA}']
+
+- id: 'Deploy to Cloud Run'
+  name: 'gcr.io/cloud-builders/gcloud:$_CLOUDSDK_VERSION'
+  args:
+  - 'run'
+  - 'deploy'
+  - '${_SERVICE}'
+  - '--image'
+  - 'gcr.io/${PROJECT_ID}/${_SERVICE_TAG}:${COMMIT_SHA}'
+  - '--allow-unauthenticated'
+  - '--region'
+  - '${_CLOUDSDK_REGION}'
+  - '--platform'
+  - 'managed'
+
+- id: 'Get Cloud Run URL'
+  name: 'gcr.io/cloud-builders/gcloud:$_CLOUDSDK_VERSION'
+  entrypoint: '/bin/bash'
+  args:
+  - '-c'
+  - |
+    get_url() {
+        gcloud run services describe ${_SERVICE} --format 'value(status.url)' \
```

Comment:
We need to have isolation for these templates: in case multiple CI runs are happening at the same time, they shouldn't have conflicts w.r.t names. In kubernetes it's easy to achieve with namespaces. Not sure how to do it with Cloud Run.

Reply:
Good callout! The current best approach is to avoid duplication by naming the service distinctly, examples would be something like:

* hello-world-$SHORT_SHA
* hello-world-$BUILD_ID

If there can be a build that runs multiple samples, we can re-introduce the language:

* hello-world-go-$SHORT_SHA
* hello-world-go-$BUILD_ID
* hello-world-go-$_PR_NUMBER

PR Number would require some extra finesse if there's anything like a nightly build of all samples.

Happy to discuss future options but this is the best practice for now. Do you have a preference?

Reply:
<lang>-hello-world-cloud-run-$short_sha 
The same with $build_id sounds good too. Up to you.

I'm not familiar with Cloud Run networking, so what happens when you want to expose your Cloud Run service to internet? Will its IP object have a name that you can manage, like service/load balancer in kubernetes? Or is it internally managed by Cloud Run?
Asking to make sure we avoid conflicts in that too.

Reply:
No need to worry about ip addresses, that's completely abstracted by the platform.

Reply:
Ended up going with COMMIT_SHA to avoid multiple substitutions. Will switch the whole thing to SHORT_SHA if something proves too long.